log/field.go

```diff
@@ -143,6 +143,21 @@ func Object(key string, obj interface{}) Field {
 	}
 }
 
+// Event adds a string-valued field with name "value" to a Span.LogFields() record
+func Event(val string) Field {
+	return String("event", val)
+}
+
+// Message adds a string-valued field with name "event" to a Span.LogFields() record
+func Message(val string) Field {
+	return String("message", val)
+}
+
+// Stack adds a string-valued field with name "stack" to a Span.LogFields() record
+func Stack(val string) Field {
+	return String("stack", val)
+}
```

Comment:
why is "stack" a string? Is the intention to represent a stack trace?

Reply:
It was declared  that way in  [semantic conventions doc](https://github.com/opentracing/specification/blob/master/semantic_conventions.md#log-fields-table) 
`| "stack" | string | A stack trace in platform-conventional format; may or may not pertain to an error. E.g., "File \"example.py\", line 7, in \<module\>\ncaller()\nFile \"example.py\", line 5, in caller\ncallee()\nFile \"example.py\", line 2, in callee\nraise Exception(\"Yikes\")\n"`

I tend to agree that this may be not suitable for some implementations, so I can drop that function.

Reply:
I think the convention is a bit misleading here. Once the tag is set and accepted by the span (e.g. when the trace is sampled), then yes, it has to be a string, as there's no other data structure available (NB: OpenTelemetry will have a proper structure for stack frames). But that doesn't mean that the tag must be set as a string, because doing so would be quite inefficient if the trace is not sampled. I always prefer the API to take native objects (e.g. https://golang.org/pkg/runtime/debug/#Stack, or a Throwable in Java) and the tracer implementation can decide to convert them to strings.

So yes, I recommend removing this, as it's not required by the rest of the PR.

Reply:
Agree. Fixed in new version.